finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/Stream.scala

```diff
@@ -68,6 +68,22 @@ object Stream {
   // performed).  This would require some form of buffering, ideally
   // in the dispatcher and server stream
 
+  /**
+   * Fail the queue and release all of its Frames.  This can be called as a safe alternative to
+   * q.fail(e, discard = true).
+   */
+  def failAndDrainFrameQueue(q: AsyncQueue[Frame], e: Throwable): Unit = {
```

Comment:
Seems like there's some duplication between this function and the `failAndDrainQueue` function in `io.buoyant.grpc.runtime.Stream`; is there any reasonable way to factor it out so we can use the same code in both places?

Reply:
Yeah, the functions are basically identical except that one operates on Frames and the other operates on Releasables.  We could consolidate them by having Frame and Releasable implement a common trait or by having `failAndDrainQueue` take a releaser function as an input.  But, given the small size of this method, I think I'd rather avoid adding an abstraction.

Reply:
Oh, I didn't realise the functions were operating on different types; carry on, then!